Mathlib/Topology/Separation/Basic.lean

```diff
@@ -725,6 +725,23 @@ theorem continuousWithinAt_congr_set' [TopologicalSpace Y] [T1Space X]
   rw [← continuousWithinAt_insert_self (s := s), ← continuousWithinAt_insert_self (s := t)]
   exact continuousWithinAt_congr_set (eventuallyEq_insert h)
 
+theorem ContinousWithinAt.eq_const_of_mem_closure [TopologicalSpace Y] [T1Space Y]
+    {f : X → Y} {s : Set X} {x : X} {c : Y} (h : ContinuousWithinAt f s x) (hx : x ∈ closure s)
+    (ht : ∀ y ∈ s, f y = c) : f x = c := by
+  rw [← Set.mem_singleton_iff, ← closure_singleton]
+  exact h.mem_closure hx ht
+
+theorem ContinousWithinAt.eq_const_of_mem_closure_of_qOn [TopologicalSpace Y] [T1Space Y]
+    {f : X → Y} {s : Set X} {x : X} {c : Y} (h : ContinuousWithinAt f s x) (hx : x ∈ closure s)
+    (ht : s.EqOn f (fun _ ↦ c)) : f x = c := by
+  apply ContinousWithinAt.eq_const_of_mem_closure h hx ht
```

Comment:
I think you can remove this version. We either stay in the `EqOn` world, or not, but let's not mix and match.

Reply:
Ok sounds good!